test/controllers/SeriesDetailsControllerSpec.scala

```diff
@@ -122,39 +135,54 @@ class SeriesDetailsControllerSpec extends FrontEndTestHelper {
     "renders an error when a valid form is submitted but there is an error from the api" in {
       val seriesId = UUID.randomUUID()
       val client = new GraphQLConfiguration(app.configuration).getClient[gs.Data, gs.Variables]()
+      val graphQLConfiguration = new GraphQLConfiguration(app.configuration)
+      val seriesService = new SeriesService(graphQLConfiguration)
+      val consignmentService = new ConsignmentService(graphQLConfiguration)
       val data: client.GraphqlData = client.GraphqlData(Option.empty, List(GraphQLClient.Error("Error", Nil, Nil, None)))
       val dataString: String = data.asJson.printWith(Printer(dropNullValues = false, ""))
       wiremockServer.stubFor(post(urlEqualTo("/graphql"))
         .willReturn(okJson(dataString)))
 
-      val controller = new SeriesDetailsController(getAuthorisedSecurityComponents, new GraphQLConfiguration(app.configuration), getValidKeycloakConfiguration)
+      val controller = new SeriesDetailsController(getAuthorisedSecurityComponents, getValidKeycloakConfiguration,
+        seriesService, consignmentService)
       val seriesSubmit = controller.seriesSubmit().apply(FakeRequest(POST, "/series").withFormUrlEncodedBody(("series", seriesId.toString)).withCSRFToken)
-      playStatus(seriesSubmit) mustBe BAD_REQUEST
+
+      seriesSubmit.failed.futureValue shouldBe a[GraphQlException]
     }
 
     "display errors when an invalid form is submitted" in {
       val client = new GraphQLConfiguration(app.configuration).getClient[gs.Data, gs.Variables]()
-      val data: client.GraphqlData = client.GraphqlData(Option.empty, List(GraphQLClient.Error("Error", Nil, Nil, None)))
+      val uuid = UUID.fromString("c2efd3e6-6664-4582-8c28-dcf891f60e68")
```

Comment:
Might be worth abstracting this value out as its used in a couple of different tests

Reply:
Done, thanks.

Thinking out loud: at some point I'd quite like to tidy these tests up a bit more. The controller and its dependencies are stateless, so we ought to be able to construct them at the top of the test rather than for each test, but it will require a bit of dependency juggling to get it to work.